lms/templates/courseware/progress_graph.js

```diff
@@ -103,18 +131,74 @@ $(function () {
     descending_grades = sorted(grade_cutoffs, key=lambda x: grade_cutoffs[x], reverse=True)
     for grade in descending_grades:
         percent = grade_cutoffs[grade]
+        ## safe-lint: disable=javascript-jquery-append
         grade_cutoff_ticks.append( [ percent, u"{0} {1:.0%}".format(grade, percent) ] )
   else:
     grade_cutoff_ticks = [ ]
   %>
   
-  var series = ${ json.dumps( series ) };
-  var ticks = ${ json.dumps(ticks) };
-  var bottomTicks = ${ json.dumps(bottomTicks) };
-  var detail_tooltips = ${ json.dumps(detail_tooltips) };
-  var droppedScores = ${ json.dumps(droppedScores) };
-  var grade_cutoff_ticks = ${ json.dumps(grade_cutoff_ticks) }
+  var series = ${ series | n, dump_js_escaped_json };
+  var ticks = ${ ticks | n, dump_js_escaped_json };
+  var bottomTicks = ${ bottomTicks | n, dump_js_escaped_json };
+  var detail_tooltips = ${ detail_tooltips | n, dump_js_escaped_json };
+  var droppedScores = ${ droppedScores | n, dump_js_escaped_json };
+  var grade_cutoff_ticks = ${ grade_cutoff_ticks | n, dump_js_escaped_json }
   
+  var yAxisTooltips={};
+
+    /*
+    series looks like:
+    [
+        {
+            color: "#600101",
+            label: "Homework",
+            data: [[1, 0.06666666666666667], [2, 1], [3.25, .53]]
+        },
+        ...
+    ]
+
+    detail_tooltips looks like:
+    {
+        "Dropped Scores": [0: "The lowest 1...:],
+        "Homework": [
+            0: "Homework 1 -- Homework -- Question Styles 7% (1/15)",
+            1: "Homework 2 -- Homework -- Get Social 100% (1/1)",
+            2: "Homework Average = 53%"
+        ],
+        ...
+    }
+    */
+
+  // loop through the series and extract the matching tick and the series label
+  for (var seriesIndex = 0; seriesIndex < series.length; seriesIndex++) {
+      for (var dataIndex = 0; dataIndex < series[seriesIndex]['data'].length; dataIndex++) {
+          var tickIndex = series[seriesIndex]['data'][dataIndex][0];
+          // There may be more than one detail tooltip for a given tickIndex. If so,
+          // push the new tooltip on the existing list.
+          if (tickIndex in yAxisTooltips) {
+              yAxisTooltips[tickIndex].push(detail_tooltips[series[seriesIndex]['label']][dataIndex]);
+          }
+          else {
+              yAxisTooltips[tickIndex] = [detail_tooltips[series[seriesIndex]['label']][dataIndex]];
+          }
+          // If this item was a dropped score, add the tooltip message about that.
+          for (var droppedIndex = 0; droppedIndex < droppedScores.length; droppedIndex++) {
+              if (tickIndex == droppedScores[droppedIndex][0]) {
```

Comment:
`===`

This makes me worried ESLint isn't linting this file - it wouldn't allow `==` if it was, but the build is green.